### DISCUSSION TITLE
问问请求是怎么跑通的

### DISCUSSION THREAD

**Discussion by BTMuli** [Arondight/Adachi-BOT#1002] [Q&A] [answered]
试了不同的请求头、不同的 salt，不同的 url，返回的除了 `不知道这样的主机` 外，就是 `invalid request`

https://github.com/BTMuli/Tauri.Genshin/issues/15

拿 cookie 来用的时候，到底用的是哪个或者哪几个 key  啊，还有就是我手头上的 `stoken` 跟 `ltoken` 怎么用啊 T_T

**Comment by Arondight** [comment]
可以跟一下 `characterDetail` 函数，但是项目中请求的数据只用到了 `cookie_token` 和 `account_id` ，不需要其他字段。

**Reply by BTMuli** [reply]
发现 `api-takumi-record` 相关请求返回的都是 `invalid request`，其他请求都没问题了，是 salt 或者其他部分不一样吗？

**Reply by mark9804** [reply]
是不是API地址改了

**Reply by BTMuli** [reply]
应该不是吧，感觉好像是参数方面少了或者 cookie 少了某项，其他项目用的也是那个地址

**Reply by BTMuli** [reply]
https://github.com/BTMuli/TGAssistant/commit/12fd7d6a9b3614e356b16880f96fe78e89c34e3f

**Reply by BTMuli** [reply]
不会被控了吧，开 inprivate 登录通行证登了就显示失效T_T

**Reply by BTMuli** [reply]
直接下了个charles抓包snap.hutao，也算是没有办法的办法了😥

**Comment by BTMuli** [comment, marked as answer]
https://github.com/DGP-Studio/Snap.Hutao/discussions/729#discussioncomment-5953032 已解决，还是谢谢 @Mark9804 老大哥了，我也不知道怎么会犯这样的错误 🤔
